alcs-frontend/src/app/shared/header/header.component.ts

```diff
@@ -12,27 +16,109 @@ import { AuthenticationService } from '../../services/authentication/authenticat
 })
 export class HeaderComponent implements OnInit {
   homeUrl = environment.homeUrl;
-  decisionMakers: ApplicationDecisionMakerDto[] = [];
+  currentUser!: ICurrentUser;
+  currentUserProfile?: UserDto;
+  private decisionMakers: ApplicationDecisionMakerDto[] = [];
+  $sortedDecisionMakers!: Observable<ApplicationDecisionMakerDto[]>;
 
   constructor(
     private authService: AuthenticationService,
     private applicationService: ApplicationService,
+    private toastService: ToastService,
+    private userService: UserService,
     private router: Router
   ) {}
 
   ngOnInit(): void {
-    this.applicationService.$applicationDecisionMakers.subscribe((dms) => {
-      this.decisionMakers = dms;
+    this.authService.isAuthenticated.subscribe((isAuthenticated) => {
+      if (isAuthenticated) {
+        this.currentUser = this.authService.getCurrentUser();
+
+        this.userService.fetchUsers();
+        this.applicationService.setup();
+      }
+    });
+
+    this.userService.$users.subscribe((users) => {
```

Comment:
Should we add a $userProfile to the userService that publishes the current users profile?